### PR TITLE
Run3-alca213X Modify many of the loops as Range based loops in Calibration/HcalCalibAlgos

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Modify many of the loops as Range ased loops in Calibration/HcalCalibAlgos as suggested by Tamas

#### PR validation:

Checked using test workflows in the test areas of Calibration/HcalAlCaRecoProducers and Calibration/HCalCalibAlgos

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by yuanchao** [comment]
+1
* to c++11 range-based for loop
* WF matrix tests and unit tests passed
* SA bugs not due to this modification

**Comment by bsunanda** [comment]
@qliphy @perrotta Could you integrate this?

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc [hunk 1]
```diff
@@ -27,7 +27,7 @@
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "CommonTools/UtilAlgos/interface/TFileService.h"
 
-//#define EDM_ML_DEBUG
+#define EDM_ML_DEBUG
```
Comment (perrotta) [inline Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc:30]: This should be commented out once again.

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc [hunk 2]
```diff
@@ -116,105 +116,107 @@ void HcalIsoTrackAnalyzer::analyze(edm::Event const& iEvent, edm::EventSetup con
...
+        t_HitEnergies = itr.hitEnergies_;
+        t_HitEnergies1 = itr.hitEnergies1_;
+        t_HitEnergies3 = itr.hitEnergies3_;
       }
 #ifdef EDM_ML_DEBUG
       if (debug)
         edm::LogVerbatim("HcalIsoTrack") << "eHcal:eHcal10:eHCal30 " << t_eHcal << ":" << t_eHcal10 << t_eHcal30;
 #endif
       tree->Fill();
+      edm::LogVerbatim("HcalIsoTrackX") << "Run " << t_Run << " Event " << t_Event << " p " << t_p;
```
Comment (perrotta) [inline Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc:219]: Shouldn't this be fenced into the `EDM_ML_DEBUG`?

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrkAnalyzer.cc [hunk 3]
```diff
@@ -953,13 +953,12 @@ std::array<int, 3> HcalIsoTrkAnalyzer::fillTree(std::vector<math::XYZTLorentzVec
                                                 const edm::Handle<reco::MuonCollection>& muonh) {
   int nSave(0), nLoose(0), nTight(0);
   //Loop over tracks
-  std::vector<spr::propagatedTrackDirection>::const_iterator trkDetItr;
+  //  std::vector<spr::propagatedTrackDirection>::const_iterator trkDetItr;
```
Comment (perrotta) [inline Calibration/HcalCalibAlgos/plugins/HcalIsoTrkAnalyzer.cc:956]: please remove this commented out line

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc (hunk 1 above)
Comment (bsunanda) [reply Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc:30]: Done

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc (hunk 2 above)
Comment (bsunanda) [reply Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc:219]: I prefer to keep it out of EDM_ML_DEBUG protection

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrkAnalyzer.cc (hunk 3 above)
Comment (bsunanda) [reply Calibration/HcalCalibAlgos/plugins/HcalIsoTrkAnalyzer.cc:956]: Done

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@perrotta @qliphy @tvami @yuanchao Please merge the PR now

**Comment by francescobrivio** [comment]
+alca

**Comment by perrotta** [comment]
+1

**Pull request merged**